test/nsidc-smap/nsidc-smap_Regression.ipynb

```diff
@@ -0,0 +1,354 @@
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "80ba1a0f-6928-4906-9002-f0a15c75f381",
+   "metadata": {},
+   "source": [
+    "## Multiple output file tests\n",
+    "\n",
+    "The following cell runs through any request that generates multiple output files. These are GeoTIFF Reformatting requests. These are generally requests for multiple variables and generate a single GeoTIFF for each variable on the output. The tests make a request and then download all of the resulting files and then compare those to the reference hash values to verify no changes in the output.\n",
+    "\n",
+    "The tests include:\n",
+    "\n",
+    "- SPL2SMP with multiple variable subsetting.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6e5e2c1c-4ad6-4761-b001-1a90af7c1bf7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if configuration is not None:\n",
+    "    with TemporaryDirectory() as tmp_dir:\n",
+    "        for test_name, test_configs in configuration[\"multiple_output_tests\"].items():\n",
+    "\n",
+    "            for shortname, test_config in test_configs.items():\n",
+    "                print(\n",
+    "                    f\"testing {shortname} with {json.dumps(test_config, indent=2, default=str)}\"\n",
+    "                )\n",
+    "                test_request = Request(**test_config[\"request_params\"])\n",
+    "                ext = test_config[\"test_params\"][\"ext\"]\n",
+    "\n",
+    "                ## Generate and download all of the output files\n",
+    "                job_id = harmony_client.submit(test_request)\n",
+    "\n",
+    "                for filename in [\n",
+    "                    file_future.result()\n",
+    "                    for file_future in harmony_client.download_all(\n",
+    "                        job_id, overwrite=True, directory=tmp_dir\n",
+    "                    )\n",
+    "                ]:\n",
+    "                    prefix = Path(filename).name.split(\"_\")[0]\n",
+    "                    print(f\"Downloaded: {Path(filename).name}\")\n",
+    "\n",
+    "                # ## Check each of the expected output files against its reference file.\n",
+    "                for test_var in test_config[\"request_params\"][\"variables\"]:\n",
+    "                    var_base = test_var.split(\"/\")[-1]\n",
+    "                    output_file = file_for_variable(\n",
+    "                        Path(tmp_dir), f\"{prefix}*Data_{var_base}_reformatted*\"\n",
+    "                    )\n",
+    "                    reference_file = file_for_variable(\n",
+    "                        Path(\"reference_files\"), f\"{var_base}_reference*\"\n",
+    "                    )\n",
+    "                    print(f\"validating {output_file.name}\")\n",
+    "                    geotiff_matches_reference_hash_file(output_file, reference_file)\n",
```

Comment:
I think this should have an `assert` before the function call. As with the other test cell, it would be good to have a message for if the assertion fails, so we know which of the specific cases being looped through failed.